From: Rob Evers <revers@redhat.com>
Date: Tue, 5 Oct 2010 14:24:45 -0400
Subject: [scsi] lpfc: ignore failure of REG_VPI mbox w/UPD bit set
Message-id: <1286288695-20754-14-git-send-email-revers@redhat.com>
Patchwork-id: 28603
O-Subject: [RHEL5.6 PATCH 13/23] Added code to ignore the failure of REG_VPI
	mailbox with UPD bit set on older tigershark firmware. (CR: 104912)
Bugzilla: 619917
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>

Added code to ignore the failure of REG_VPI mailbox with UPD bit set on older tigershark firmware. (CR: 104912)

From: Rob Evers on behalf of Emulex <revers@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=619917

diff --git a/drivers/scsi/lpfc/lpfc_els.c b/drivers/scsi/lpfc/lpfc_els.c
index 1df2264..0cf47e9 100644
--- a/drivers/scsi/lpfc/lpfc_els.c
+++ b/drivers/scsi/lpfc/lpfc_els.c
@@ -6543,8 +6543,12 @@ lpfc_cmpl_reg_new_vport(struct lpfc_hba *phba, LPFC_MBOXQ_t *pmb)
 
 	if (mb->mbxStatus) {
 		lpfc_printf_vlog(vport, KERN_ERR, LOG_MBOX,
-				 "0915 Register VPI failed: 0x%x\n",
-				 mb->mbxStatus);
+				"0915 Register VPI failed : Status: x%x"
+				" upd bit: x%x \n", mb->mbxStatus,
+				 mb->un.varRegVpi.upd);
+		if (phba->sli_rev == LPFC_SLI_REV4 &&
+			mb->un.varRegVpi.upd)
+			goto mbox_err_exit ;
 
 		switch (mb->mbxStatus) {
 		case 0x11:	/* unsupported feature */
@@ -6603,7 +6607,7 @@ lpfc_cmpl_reg_new_vport(struct lpfc_hba *phba, LPFC_MBOXQ_t *pmb)
 			lpfc_do_scr_ns_plogi(phba, vport);
 
 	}
-
+mbox_err_exit:
 	/* Now, we decrement the ndlp reference count held for this
 	 * callback function
 	 */
